### ISSUE TITLE
infogan code cannot generate reasonable result

### ISSUE THREAD

**Issue by kudo1026** [burness/tensorflow-101#13]
Hi,

I'm running the original code you give for infogan. I used `train.py` to train the network first, and then used `generate.py` to generate sample pictures. However, the result picture does not give the expected result. Both `sample.png` and `fake.png` are all pictures of number 1, which shows that the categorical latent code doesn't work. Also, for the two continuous latent code, they don't give very intuitive meaning, but rather like random noise.

I noticed that during training, the generation loss remains pretty large even at the end of 50 training epochs. It's like this:
`cur epoch 50 update l_d step 85900, loss_disc 0.0126310065389, loss_gen 5.10504961014`

I wonder did you ever test on it? What kind of results can you get? Or do you have any idea how to correct this?

Thanks in advance
